src/robot/libraries/BuiltIn.py

```diff
@@ -2789,8 +2789,10 @@ def set_test_message(self, message, append=False):
 
         In test teardown this keyword can alter the possible failure message,
         but otherwise failures override messages set by this keyword. Notice
-        that in teardown the initial message is available as a built-in variable
-        ``${TEST MESSAGE}``.
+        that in teardown the test message is available as a built-in
+        variable ``${TEST MESSAGE}`` which will be equal to the initial test
+        message at the start of the test teardown (before the first use of
+        this keyword).
```

Comment:
I agree the original explanation is not good but the proposed one is also pretty confusing. Would it be enough to just drop `initial` from the original doc?